client/my-sites/stats/stats-module/connected-list.js

```diff
@@ -0,0 +1,132 @@
+/**
+ * External dependencies
+ */
+import React, { PropTypes, Component } from 'react';
+import { connect } from 'react-redux';
+import classNames from 'classnames';
+
+/**
+ * Internal dependencies
+ */
+import ErrorPanel from '../stats-error';
+import StatsModuleExpand from './expand';
+import StatsList from '../stats-list';
+import StatsListLegend from '../stats-list/legend';
+import DatePicker from '../stats-date-picker';
+import Card from 'components/card';
+import StatsModulePlaceholder from './placeholder';
+import SectionHeader from 'components/section-header';
+import QuerySiteStats from 'components/data/query-site-stats';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { getSiteSlug } from 'state/sites/selectors';
+import {
+	isRequestingSiteStatsForQuery,
+	getSiteStatsNormalizedData
+} from 'state/stats/lists/selectors';
+
+class StatsConnectedModule extends Component {
+	static propTypes = {
+		summary: PropTypes.bool,
+		moduleStrings: PropTypes.object,
+		period: PropTypes.object,
+		path: PropTypes.string,
+		siteSlug: PropTypes.string,
+		siteId: PropTypes.number,
+		date: PropTypes.string,
+		data: PropTypes.array,
+		query: PropTypes.object,
+		statType: PropTypes.string,
+		showSummaryLink: PropTypes.bool
+	};
+
+	static defaultProps = {
+		showSummaryLink: false,
+		query: {}
+	};
+
+	getModuleLabel() {
+		if ( ! this.props.summary ) {
+			return this.props.moduleStrings.title;
+		}
+
+		return ( <DatePicker period={ this.props.period.period } date={ this.props.period.startOf } summary={ true } /> );
+	}
+
+	getHref() {
+		const { summary, period, path, siteSlug, date } = this.props;
+
+		// Some modules do not have view all abilities
+		if ( ! summary && period && path && siteSlug ) {
+			return '/stats/' + period.period + '/' + path + '/' + siteSlug + '?startDate=' + date;
+		}
+
+		return null;
```

Comment:
Minor and maybe arguable to have the explicitness, but you could omit the fallback `return`, as the default return value `undefined` would work just as well for your use case).